### PR TITLE
Can we migrate to TS gradually?

### PR DESCRIPTION
**Pull request by gustavo-salazar**
This is a proof of concept creating a setup that builds the website using its existing codebase but accepting `.ts` and `.tsx` files. It also includes changes to start using the visual-framework and gradually remove any foundation usage.

The proposed TS setup:
* In [webpack.config.js](https://github.com/ProteinsWebTeam/interpro7-client/blob/97589eeafef11a78b995b792e3de9d11d4d46432/webpack.config.js) is to simply process TypeScript files with `ts-loader` and then apply the `babel-loader` with the same config as with JavaScripts files.
* The type checking when building in webpack is ran via `fork-ts-checker-webpack-plugin`
* FlowJS in the meantime will consider `.ts` and `.tsx` files as empty objects.
* `eslint` will be run twice. The old way for JS files, and a new config with `typescript-eslint`
* `jest` needed a new preset, but otherwise is the same setup. In this iteration, I'm not migrating the tests into TS.
* I'm using `git mv` to conserve the history of the migrated files.
* I'm adding `// $FlowFixMe` when importing a TS file in a JS one, to avoid flow errors. Once the file is migrated to TS, I remove the comments.
* When migrating a file I use the info we have in the file from `propTypes` and FlowJS, but remove both leaving all typing checks to TS.
* `tsconfig.json` was updated and among other things now it runs on `strict` mode.

Changes for the visual framework.
* Including the bundled vs framework from EMBL CDN.
* Replacing the foundationPartial function for `cssBinder` on `.tsx` files. The new function has similar logic to extend `classNames` and allow binding multiple css modules. The difference is that it doesn't include `foundation-float.css` and is slightly improved to avoid replication of class names.
* For each migrated component I'm replacing foundation classes such `row` and `column` for similar ones in VF such as `vf-stack` of `vf-grid`
* Also trying to remove usage of [interpro-new.css](https://github.com/ProteinsWebTeam/interpro7-client/blob/97589eeafef11a78b995b792e3de9d11d4d46432/src/styles/interpro-new.css) and adding anything that is actually been used from it in [interpro-vf.css](https://github.com/ProteinsWebTeam/interpro7-client/blob/97589eeafef11a78b995b792e3de9d11d4d46432/src/styles/interpro-vf.css).

The components used as proof of context were all the ones used under `src/components/Entry/Summary`, however, the major complexity comes from the migration of [loadData](https://github.com/ProteinsWebTeam/interpro7-client/blob/97589eeafef11a78b995b792e3de9d11d4d46432/src/higherOrder/loadData/index.js). In this case, I'm leaving 2 versions the traditional in JS, and a new on in TS that deals with all the Generic types involved in it.

### CODE REVIEW

**Review by matthiasblum — APPROVED** [review]
Looks good to me. I made a few comments or suggestions, but nothing major.
One question, though: the Visual Framework is imported from the EMBL CDN. Will it stay like this, or will `vf-core` be added to the dependencies?

File: src/components/Description/Paragraph/index.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                target="_blank"
+                className={css('ext')}
+                key={i}
+              >
+                {tagValue}
+              </Link>
+            );
+          }
+        }
+        // TODO: change the way descriptions work from the backend side.
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply src/components/Description/Paragraph/index.tsx:132]: Yoo, I dig into the history of this line, it's at least 5 years old. 

My best guess is about the whole thing that the description is an array of fragments. I think we once considered the idea of simply having a string in there, but that requires too many changes in different places and never got actually done.

I'll remove the TODO

File: src/components/Description/Citations/index.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                  db: customLocation.description[key].db,
+                  accession,
+                },
+              },
+              hash: pubId,
+            };
+          }}
+        >
+          {refCounter}
+          {i + 1 < text.split(',').length && ', '}
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply src/components/Description/Citations/index.tsx]: 🤓 TIL `.map` had the array as a third parameter

File: src/components/Entry/Summary/MemberDBSubtitle/index.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                        db: metadata.source_database,
+                        accession: metadata.set_info.accession,
+                      },
+                    },
+                  }}
+                >
+                  {metadata.set_info.name}
+                </Link>
+              ) : (
+                'ø'
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply src/components/Entry/Summary/MemberDBSubtitle/index.tsx:84]: This shouldn't happen as the counter should be `0` if there is not `set_info` but I guess an extra failover there is not much of an issue, in the weird case of corrupted data.

**Pull request merged**